mutt/memory.h

```diff
@@ -23,6 +23,9 @@
 #ifndef MUTT_MUTT_MEMORY_H
 #define MUTT_MUTT_MEMORY_H
 
+#if __has_include(<stdcountof.h>)
```

Comment:
I think we might need to guard the whole thing by `#if defined __has_include`.

Reply:
> I think we might need to guard the whole thing by `#if defined __has_include`.

I don't think that's valid.  :)

Reply:
> Or check for the header file in our configure script, like we do for other headers

Hmmm, okay, that's a valid thing to do.

On the other hand, which compilers do we support?  `__has_include` is standard ISO C23, and has been supported by compilers for a long-ish time:

<https://en.cppreference.com/w/c/compiler_support.html>

-  GCC 5 (2015)
-  Clang (forever?)
-  MSVC 19.11 (2017)
-  Intel C++ 18.0 (2017)

Reply:
> > > I think we might need to guard the whole thing by `#if defined __has_include`.
> > 
> > 
> > I don't think that's valid. :)
> 
> I think it is: https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

Hmmm, interesting.  I don't know if the other implementations behave like that, but it could work.

Reply:
Done, in v2.  Thanks!